network-area-diagram-viewer/src/main/java/com/powsybl/nad/viewer/DiagramController.java

```diff
@@ -91,9 +99,53 @@ public void createDiagram(Model model, StringProperty modelSvgContent, Container
         });
         svgContent.textProperty().bind(modelSvgContent);
 
+        // Link the jsHandler with this controller
+        jsHandler.linkWithController(model, modelSvgContent, container);
+
         updateDiagram(model, modelSvgContent, container);
     }
 
+    public static void updateDiagram(Model model, StringProperty modelSvgContent, Container<?> container, Map<String, Point> postLayoutPositions) {
+        Objects.requireNonNull(postLayoutPositions);
+        // TODO(Luma) not the best way to do it, just to explore how things could work ...
+        // Instead of updating only the location of the given equipment
+        // We do a complete layout without restrictions,
+        // then apply the given post layout positions to the result
+        // and after that we use the updated positions to do final complete drawing
+        StringWriter writer = new StringWriter();
+        Service<String> nadService = new Service<>() {
+            @Override
+            protected Task<String> createTask() {
+                return new Task<>() {
+                    @Override
+                    protected String call() {
+                        NetworkAreaDiagram nad = getNetworkAreaDiagram(model, container);
+                        // First layout without restrictions
+                        Map<String, Point> positions = nad.layout(model.getLayoutParameters(), model.getLayoutFactory());
```

Comment:
I wouldn't add that `NetworkAreaDiagram::layout` method then. If we save all positions, there's no need for it.

Of course we still need to get those positions to save them first, but we'd better read them from the SVG. Indeed, the frond end will need to get those positions, and it gets only the SVG from the back end. 

We could read that info either from the nodes translate value or from the metadata (wondering if should we add that info there?)

Reply:
With the ::layout method I was trying to separate the layout itself from its visualization, at the API level. The layout, without building the complete visualization may be interesting in other use cases (building a diagram in stages, or preparing a totally different visualization not based in SVG...). 

I agree with your point, here I just used the method to simplify getting again the initial coordinates of all the elements without having to pass them around.

Reply:
ok, got it! Shouldn't we rather add this in the Layout interface then? As the return value of `Layout::run`

Reply:
Updated the interaction demonstration following your suggestions: the client gathers _all_ equipment positions from SVG and sends a request to build an updated diagram with those positions.

Information about positions between client and server is exchanged through JSON. No need for specific serialization for now.